libs/services/src/dune-simulator/dune-simulator.service.ts

```diff
@@ -22,8 +23,7 @@ export class DuneSimulatorService {
         }
 
         try {
-            const formattedHeaders = body.schema.map(header => header.name).join(',');
-            await this.csvFileRepository.createTable(body.table_name, formattedHeaders);
+            await this.dynamicCollectionRepository.createTable(body.table_name, body.schema);
```

Comment:
we should check (here in the simulator, but also on the real dune service) the field names. They expect their names to be separated using snake case (so for example: `borrowedAmount` -> `borrowed_amount`). Therefore, we can update the field names from camel case (if any) to snake case. 
ref: https://www.geeksforgeeks.org/how-to-convert-camel-case-string-to-snake-case-in-javascript/